### ISSUE TITLE
Quality of life improvement: Skip already saved room Data

### ISSUE THREAD

**Issue by GogetaCross** [stoofin/roomviewer-rde#2]
The program itself terminates often, especially when extracting all PC room data.
Would it be possible to add a simple check if one or more files are already saved in the traget folder and if yes just skips them?

**Comment by stoofin** [comment]
Yeah I think that'd be possible, but I'm not sure what exactly the problem is so I don't know if you need it to skip per-room or per-layer or what.

It might be easier to fix the crashing.  Could you try running it from the command prompt or powershell and see if it outputs any clue as to why it's terminating?  Maybe an uncaught exception?

**Comment by GogetaCross** [comment]
What's the cmd prompt for extaracting normal as well as HD room files to a specific folder? :D
Btw. I'm the modder that made the Crossification Mod (thanks for your tool btw.) found here: https://www.nexusmods.com/chronocrosstheradicaldreamersedition/mods/61

I want to release some scripts that will help future Background modders with extracting and layering the room files and later on to combine them back so it's readbale and also importable/convertable to chip2 file using your viewer. Is it okay if I pack your tool together with the files to distribute (giving you full credits of course) as further changes/updates to your tool can prevent the insertion to be incorrect like changing how rooms are loaded/saved.

**Comment by stoofin** [comment]
Oh, it doesn't really do command line arguments, but if you launch it from the command prompt you should be able to see any errors it writes out within the command prompt window.

Good job on the mod!

I think some scripts to help is a great idea.  *I* have no problem with you including this in your pack, but this tool is just a modified version of a tool available from https://www.chronocompendium.com/Term/Utilities_(PSX).html so I'd say credit the original author, Magil (whose toes I in turn hope I'm not stepping on by putting up this version).  You can also point people here if they want to report any bugs.

**Comment by GogetaCross** [comment]
Thanks a lot! :D

If I try to run the program with normal cmd then I don't see anything if the program crashes. If I use PowerShell it prompts Visual Studio to open for debug. If I allow it it says this "System.AccessViolationException: 'An attempt was made to read or write in the protected memory. This is often an indication that other memory is corrupted."

It's tanslated from German, so take it with a grain of salt.
Maye something with the viewer.sln?

**Comment by stoofin** [comment]
Thanks for finding that error message, it's better than nothing but doesn't pinpoint the problem like I'd hoped.  Which operation exactly is crashing?  Export All Rooms?

I double-checked and it still manages to export all rooms in both HD and native res correctly and without crashing for me, using unmodified cdrom.dat and hd.dat from steam (which I copied together to another folder).

Anyway, I made a version that added that checkbox you suggested.  It skips exporting rooms that already have any existing layers in the target directory (at the risk of potentially leaving rooms partially exported after a crash, I guess?).  I also put in a status bar during export so that if it skips everything you can see it quickly iterate through the rooms, if only to update the label.

Here it is ([source](https://github.com/stoofin/roomviewer-rde/tree/skip-existing)):
[viewer-with-skip.zip](https://github.com/stoofin/roomviewer-rde/files/13255975/viewer-with-skip.zip)
Let me know if it's what you needed and if it works, in which case I can put it under releases.

Eventually I'd like to try running it on windows and see if I can reproduce your crash.
